packages/context/src/inject.ts

```diff
@@ -176,11 +185,10 @@ export function describeInjectedArguments(
   target: any,
   method?: string | symbol,
 ): Injection[] {
-  if (method) {
-    return Reflector.getMetadata(PARAMETERS_KEY, target, method) || [];
-  } else {
-    return Reflector.getMetadata(PARAMETERS_KEY, target) || [];
-  }
+  method = method || '';
+  const meta = Reflector.getMetadata(PARAMETERS_KEY, target);
```

Comment:
IIUC, `describeInjectedArguments` is reading metadata set by `@inject`, which uses `ParameterDecoratorFactory.createDecorator` internally.

I'd like this code to be decoupled from `Reflector` and use a helper method from `@loopback/metadata` to read the information stored by `ParameterDecoratorFactory.createDecorator`.

Reply:
I think `MetadataInspector` is the tool to use here?